src/daemon/daemon.cpp

```diff
@@ -1504,6 +1594,40 @@ catch (const std::exception& e)
     return grpc::Status(grpc::StatusCode::FAILED_PRECONDITION, e.what(), "");
 }
 
+grpc::Status mp::Daemon::restart(grpc::ServerContext* context, const RestartRequest* request,
+                                 grpc::ServerWriter<RestartReply>* server) // clang-format off
+try // clang-format on
+{
+    mpl::ClientLogger<RestartReply> logger{mpl::level_from(request->verbosity_level()), *config->logger, server};
+
+    auto instances_and_status =
+        find_requested_instances(request->instance_names().instance_name(), vm_instances, deleted_instances);
+    const auto& instances = instances_and_status.first; // use structured bindings instead in C++17
+    auto& status = instances_and_status.second;         // idem
+
+    if (status.ok())
+    {
+        status = cmd_vms(instances,
+                         std::bind(&Daemon::reboot_vm, this, std::placeholders::_1)); // 1st pass to reboot all targets
+
+        if (status.ok())
+        {
+            status = cmd_vms(instances, [this](auto& vm) {
```

Comment:
clang on the Mac build fails:
```
../src/daemon/daemon.cpp:1615:42: error: lambda capture 'this' is not used [-Werror,-Wunused-lambda-capture]
            status = cmd_vms(instances, [this](auto& vm) {
```

Reply:
I was having a look just now. It's a leftover capture from an earlier version. Will remove it now.